File: 프로그래머스/0/181942. 문자열 섞기/문자열 섞기.cs
using System;

public class Solution
{
    public string solution(string str1, string str2)
    {
        int index1 = 0;
        int index2 = 0;
        bool _isStr1 = true;
        bool _isStr2 = false;
        string answer = "";
        
        while (index1 < str1.Length || index2 < str2.Length)
        {
            if (_isStr1)
            {
                answer += str1[index1++];
                _isStr1 = false;
                _isStr2 = true;
            }
            else if (_isStr2)
            {
                answer += str2[index2++];
                _isStr2 = false;
                _isStr1 = true;
            }
        }
        return answer;
    }
}

File: 프로그래머스/0/181942. 문자열 섞기/문자열 섞기.cpp
#include <iostream>
#include <string>

using namespace std;

string solution(string str1, string str2);

int main(void) {

    string str1, str2;

    std::cout << "첫번째 변수입력 : ";
    std::cin >> str1;
    std::cout << "두번째 변수입력 : ";
    std::cin >> str2;

    solution(str1, str2);

    return 0;
}

string solution(string str1, string str2) {
    string answer = "";

    for (int i = 0; i < str1.size(); i++)
    {
        answer += str1[i];
        answer += str2[i];
    }
    return answer;
}